hilti/runtime/include/exception.h

```diff
@@ -66,7 +92,7 @@ class Exception : public std::runtime_error {
 
     std::string _description;
     std::string _location;
-    Backtrace _backtrace;
+    std::unique_ptr<Backtrace> _backtrace; // null if not available
```

Comment:
Unless there is a clear performance win when going with allocating this on the heap I'd say go with `std::optional` instead here. That would avoid having to explicitly deal with `std::unique_ptr` not being copyable which IMO is not the semantics we want for an `Exception`.